### DISCUSSION TITLE
[STILL DISCUSSING IN COMMENTS] Submit Triggers Separate Forms

### DISCUSSION THREAD

**Discussion by crimsonpython24** [react-hook-form/react-hook-form#5432] [Q&A] [answered]
**Describe the bug**
When I pressed a "submit" button in either of the two forms in the CodeSandBox reproduction, the other form will be triggered.

**To Reproduce**

1. Go to the CSB instance below
1. Click on the blue "Change Password" button without typing anything. You should see a yellow border in both boxes.
2. Open the modal in the "Delete Account" button. The field should also be triggered a warning.
3. Type anything in either three fields to clear the errors.
1. Click the "Delete Account" button. Also press "OK" without typing anything. You should see an error.
2. Press [ESC] to exit the modal. You should also see that the two background input boxes were highlighted.

**Codesandbox link (Required)**
- [JS](https://codesandbox.io/s/condescending-haibt-eqef5?file=/src/App.js) Template

**Expected behavior**
In both examples above, you could see that both forms were highlighted whenever I pressed either submit buttons ("Change Password" for the first form and "OK" for the second).

I do not want that -- the "Change Password" button should only submit the first form, while "OK" should only submit the second.

**Screenshots**
![image](https://user-images.githubusercontent.com/76871022/120315987-83e9de80-c30f-11eb-843c-d8e296099d79.png)

**Desktop (please complete the following information):**

- OS: Windows Build 19041
- Browser: Edge 91.0.864.37&nbsp;(Official build)&nbsp;(64-bit)

**Comment by bluebill1049** [comment]
do not nest `form` inside a `form`. 

```
<form>
 <form />
</form>
```

to 

```
<form />
<form />
```

**Reply by crimsonpython24** [reply]
The two forms are already separated, no?

```jsx
<Form onFinish={handleSubmit(onSubmit)}>
</Form>

<div>
<CollectionCreateForm
  {...props}
/>
</div>
```

**Reply by bluebill1049** [reply]
are they have a seperate `useForm`?

**Reply by crimsonpython24** [reply]
As a matter of fact, `CollectionCreateForm` is not a form itself. The "real" form is within the CCF component. The other form is parallel to the `<div>` tag that CCF is resting in, as the code example above.

Please correct me if I'm wrong: https://codesandbox.io/s/boring-fast-qww2g?file=/src/App.js

**Reply by crimsonpython24** [reply]
> are they have a seperate `useForm`?

How do I use a separate `useForm`?

Currently, I have:
```jsx

  const {
    register,
    formState: { errors },
    handleSubmit,
    clearErrors,
    setValue,
    getValues
  } = useForm();
```

But, I believe that both forms seemed to derive from the same `useForm();`. How can I change it?

**Reply by bluebill1049** [reply]
I have a feeling the button click even got propagated to the parent.

**Reply by crimsonpython24** [reply]
> I have a feeling the button click even got propagated to the parent.

Can you please elaborate on that?

How could that cause both forms to be triggered at once?

**Reply by crimsonpython24** [reply]
Sorry if the Ant Design props bothered you, by the way. If there's anything that needs to be changed, please also include them in your reply.

**Reply by crimsonpython24** [reply]
I ended up creating aliases for the two form objects, something like:

```jsx
const { register, trigger, formState: { errors }, setError, handleSubmit, clearErrors , setValue, getValues } = useForm();
  const {
    register: register2,
    trigger: trigger2,
    formState: { errors: errors2 },
    setError: setError2,
    handleSubmit: handleSubmit2,
    clearErrors: clearErrors2,
    setValue: setValue2,
    getValues: getValues2 
  } = useForm();
```

Thanks for your help!

**Reply by thanh-nguyen-95** [reply]
I believe the reason it didn't work is because there's only 1 `handleSubmit` and it was called both when pressing `Change password` and `Ok`.

Creating 2 `useForm` instances should fix the problem.

**Reply by crimsonpython24** [reply]
How do I create two `useForm()` instances? @thanh-nguyen-95

**Reply by thanh-nguyen-95** [reply]
> 
> 
> I ended up creating aliases for the two form objects, something like:
> 
> ```js
> const { register, trigger, formState: { errors }, setError, handleSubmit, clearErrors , setValue, getValues } = useForm();
>   const {
>     register: register2,
>     trigger: trigger2,
>     formState: { errors: errors2 },
>     setError: setError2,
>     handleSubmit: handleSubmit2,
>     clearErrors: clearErrors2,
>     setValue: setValue2,
>     getValues: getValues2 
>   } = useForm();
> ```
> 
> Thanks for your help!

You already did

**Comment by crimsonpython24** [comment, marked as answer]
https://github.com/react-hook-form/react-hook-form/discussions/5432#discussioncomment-810578
